util/topsql/reporter/reporter.go

```diff
@@ -209,14 +229,91 @@ func (tsr *RemoteTopSQLReporter) Close() {
 	tsr.client.Close()
 }
 
+func addEvictedCPUTime(collectTarget map[string]*dataPoints, timestamp uint64, totalCPUTimeMs uint32) {
+	if totalCPUTimeMs == 0 {
+		return
+	}
+	others, ok := collectTarget[keyOthers]
+	if !ok {
+		others = &dataPoints{}
+		collectTarget[keyOthers] = others
+	}
+	if len(others.TimestampList) == 0 {
+		others.TimestampList = []uint64{timestamp}
+		others.CPUTimeMsList = []uint32{totalCPUTimeMs}
+	} else {
+		others.TimestampList = append(others.TimestampList, timestamp)
+		others.CPUTimeMsList = append(others.CPUTimeMsList, totalCPUTimeMs)
+		length := len(others.TimestampList)
+		if others.TimestampList[length-1] < others.TimestampList[length-2] {
+			// This situation appears only when time jump backward.
+			// If so, sort the data points by timestamp.
+			sort.Sort(others)
+		}
+	}
+	others.CPUTimeMsTotal += uint64(totalCPUTimeMs)
+}
+
+func (d *dataPoints) merge(other *dataPoints) {
+	if other == nil || len(other.TimestampList) == 0 {
+		return
+	}
+	if other.isInvalid() {
+		logutil.BgLogger().Warn("[top-sql] data points is invalid, it should never happen", zap.Reflect("self", *d), zap.Reflect("other", *other))
+		return
+	}
+	if !sort.IsSorted(other) {
+		// This situation appears only when time jump backward.
+		// If so, sort the data points by timestamp.
+		sort.Sort(other)
+	}
+	if len(d.TimestampList) == 0 {
+		d.TimestampList = other.TimestampList
+		d.CPUTimeMsList = other.CPUTimeMsList
+		d.CPUTimeMsTotal = other.CPUTimeMsTotal
```

Comment:
How about `*d = *other`?

Reply:
No, this will also assign value to `SQLDigest` and `PlanDigest`.